### PR TITLE
Exo Test: exit with status 1 on failures

### PR DESCRIPTION
**Pull request by mcharawi**

### CODE REVIEW

File: src/cmd/test.go [hunk 1]
```diff
@@ -71,7 +71,7 @@ var testCmd = &cobra.Command{
 				panic(err)
 			}
 		}
-
+		os.Exit(0)
```
Comment (charlierudolph) [inline src/cmd/test.go]: Is this needed? I think the default exit code is 0

File: test_helpers/shared_feature_context.go [hunk 2]
```diff
@@ -201,6 +201,8 @@ func SharedFeatureContext(s *godog.Suite) {
 			} else {
 				return fmt.Errorf("cmd.Wait: %v", err)
 			}
+		} else {
+			return nil
```
Comment (charlierudolph) [inline test_helpers/shared_feature_context.go]: Shouldn't this branch validate that the expected exit code is 0?

**Review by charlierudolph — COMMENTED** [review]
Please also add exit code validation to the other features in `features/test`

**Comment by mcharawi** [comment]
@charlierudolph comments addressed

File: src/application/tester.go [hunk 3]
```diff
@@ -55,7 +56,9 @@ func (a *Tester) RunAppTests() error {
 	if numFailed == 0 {
 		return a.Logger.Log("exo-test", "All tests passed", true)
 	}
-	return a.Logger.Log("exo-test", fmt.Sprintf("%d tests failed", numFailed), true)
+	err := a.Logger.Log("exo-test", fmt.Sprintf("%d tests failed", numFailed), true)
+	os.Exit(1)
```
Comment (charlierudolph) [inline src/application/tester.go]: Please no exiting from helpers. We should only have exit statements in the `cmd` directory. Please update these methods to return a bool of whether or not the tests ran successfully.

**Comment by mcharawi** [comment]
@charlierudolph comments addressed

File: src/application/tester.go [hunk 4]
```diff
@@ -53,21 +53,23 @@ func (a *Tester) RunAppTests() error {
...
 }
 
 // RunServiceTest runs the tests for a single service
-func (a *Tester) RunServiceTest(serviceName string) error {
+func (a *Tester) RunServiceTest(serviceName string) (bool, error) {
+	testsPassed := true
 	if a.InternalServiceConfigs[serviceName].Tests == "" {
 		a.logChannel <- fmt.Sprintf("%s has no tests, skipping", serviceName)
 	} else {
 		if _, err := a.runServiceTests(serviceName, a.InternalServiceConfigs[serviceName]); err != nil {
```
Comment (charlierudolph) [inline src/application/tester.go]: `runServiceTests` returns whether or not service tests pass. I believe we need to use that value. If you add a test case for running the tests for a single service where the tests fail, I believe you would currently get an exit code of 0 when we want 1.

**Comment by mcharawi** [comment]
@charlierudolph comments addressed

**Review by charlierudolph — APPROVED** [review]

**Pull request merged**